Feature/add unmanaged protocol for types that prefer a unique id other than object

roanutil opened the pull request:
Major refactor of the v3.0 preview code

roanutil committed:
Major refactor

feature/add-unmanaged-protocol-for-types-that-prefer-a-unique-id-other-than-ObjectID

roanutil committed:
Only include benchmark dependency and target if ENV enables it

feature/add-unmanaged-protocol-for-types-that-prefer-a-unique-id-other-than-ObjectID

roanutil committed:
Run swiftformat

feature/add-unmanaged-protocol-for-types-that-prefer-a-unique-id-other-than-ObjectID

roanutil committed:
Revert temporary swift version and supported platform changes

feature/add-unmanaged-protocol-for-types-that-prefer-a-unique-id-other-than-ObjectID

roanutil committed:
Revert Package.resolved

feature/add-unmanaged-protocol-for-types-that-prefer-a-unique-id-other-than-ObjectID

roanutil committed:
Update dependencies

feature/add-unmanaged-protocol-for-types-that-prefer-a-unique-id-other-than-ObjectID

roanutil committed:
Update workflow platform matrix

feature/add-unmanaged-protocol-for-types-that-prefer-a-unique-id-other-than-ObjectID

roanutil committed:
Re-bump minimum swift version to 5.10

feature/add-unmanaged-protocol-for-types-that-prefer-a-unique-id-other-than-ObjectID

r-jarvis commented on Sources/CoreDataRepository/CoreDataRepository+Aggregate.swift:
Getting `Sendable` warnings on the `subscription.cancel()` methods here. Could we make `AggregateSubscription<Value>` conditionally conform to `Sendable` only where `Value` is `Sendable`?

Haven't tested this, so there may be a good reason not to.

roanutil replied:
There's really no path to making the subscription types Sendable in their current form. I have a separate branch where that's being worked on but I would say it's outside the scope of this PR.

r-jarvis commented on Sources/CoreDataRepository/CoreDataRepository+Create.swift:
```diff
@@ -0,0 +1,39 @@
+    public func create<Model>(
+        _ item: Model,
+        transactionAuthor: String? = nil
+    ) async -> Result<Model, CoreDataError> where Model: WritableUnmanagedModel, Model: FetchableUnmanagedModel {
```
Curious, any reason to use the syntax `Result<Model, CoreDataError> where Model: WritableUnmanagedModel, Model: FetchableUnmanagedModel` over `Result<Model, CoreDataError> where Model: WritableUnmanagedModel & FetchableUnmanagedModel`? I thought the second method was more concise and commonly used in Swift.

roanutil replied:
Personally, I prefer the single conformance per clause format. I think this format is more consistently formattable and readable because each clause is the same shape instead of some having two conformances separated by `&`.

r-jarvis commented on Sources/Internal/ModelsWithIntId/FetchableModel_Int.swift:
Is there a reason a `package static func seeded(_ seed: Int) -> Self` method does not exist for this Model?

roanutil replied:
Just forgot to add it. Fixed.

roanutil committed:
Fix missing methods for FetchableModel_Int

feature/add-unmanaged-protocol-for-types-that-prefer-a-unique-id-other-than-ObjectID
